site/utils/apiUtils/index.ts

```diff
@@ -107,3 +115,205 @@ export const getReturnPage = (req: NextApiRequest) => {
+                        Authorization: authToken,
+                    },
+                });
+
+                if (responseToken.ok) {
+                    const tokenResult = hootsuiteTokenSchema.parse(await responseToken.json());
+                    const key = refreshToken?.Name || "";
+                    await putParameter(key, tokenResult.refresh_token ?? "", "SecureString", true);
+
+                    let imageLink = { id: "", url: "" };
+                    let canUpload = false;
+
+                    if (socialMediaPost.image) {
+                        const responseImage = await fetch(`${HOOTSUITE_URL}v1/media`, {
+                            method: "POST",
+                            body: JSON.stringify({
+                                sizeBytes: socialMediaPost.image.size,
+                                mimeType: socialMediaPost.image.mimetype,
+                            }),
+                            headers: {
+                                "Content-Type": "application/json",
+                                Authorization: `Bearer ${tokenResult.access_token ?? ""}`,
+                            },
+                        });
+
+                        if (responseImage.ok) {
+                            const image = hootsuiteMediaSchema.parse(await responseImage.json());
+                            const imageContents = await readFile(socialMediaPost.image.filepath || "");
+                            imageLink = { url: image.data.uploadUrl, id: image.data.id };
+
+                            const uploadResponse = await fetch(imageLink.url, {
+                                method: "PUT",
+                                headers: {
+                                    "Content-Type": socialMediaPost.image.mimetype,
+                                },
+                                body: imageContents,
+                            });
+                            if (uploadResponse.ok) {
+                                for (let i = 0; i < 3; i++) {
+                                    const imageStatus = await fetch(`${HOOTSUITE_URL}v1/media/${imageLink.id}`, {
+                                        method: "GET",
+                                        headers: {
+                                            Authorization: `Bearer ${tokenResult.access_token ?? ""}`,
+                                        },
+                                    });
+                                    if (imageStatus.ok) {
+                                        const imageState = hootsuiteMediaStatusSchema.parse(await imageStatus.json());
+
+                                        if (imageState.data.state === "READY") {
+                                            canUpload = true;
+
+                                            break;
+                                        } else {
+                                            await delay(1000);
```

Comment:
is this required as if the image state is not ready then we are waiting for 5s below